src/internal/m365/exchange/events_restore.go

```diff
@@ -210,15 +212,108 @@ func updateExceptionOccurrences(
 
 		evt = toEventSimplified(evt)
 
-		// TODO(meain): Update attachments (might have to diff the
-		// attachments using ids and delete or add). We will have
-		// to get the id of the existing attachments, diff them
-		// with what we need a then create/delete items kinda like
-		// permissions
 		_, err = ac.PatchItem(ctx, userID, ptr.Val(evts[0].GetId()), evt)
 		if err != nil {
 			return clues.Wrap(err, "updating event instance")
 		}
+
+		evt, err = api.EventFromMap(instance)
+		if err != nil {
+			return clues.Wrap(err, "parsing exception event (second)")
+		}
+
+		err = updateAttachments(ctx, ac, userID, containerID, ptr.Val(evts[0].GetId()), evt)
+		if err != nil {
+			return clues.Wrap(err, "updating attachments")
+		}
+	}
+
+	return nil
+}
+
+func updateAttachments(
+	ctx context.Context,
+	client api.Events,
+	userID, containerID, eventID string,
+	event models.Eventable,
+) error {
+	// Attachment can only be deleted or added in modified events and
+	// when an event is modified, the attachment ids will be
+	// changed. This change is both present in the data that we backed
+	// up and so we cannot rely on the attachment id to decide if they
+	// are the same. Instead we make use of the contentBytes and name
+	// to ensure they are the same.
+	attachments, err := client.GetAttachments(ctx, false, userID, eventID)
+	if err != nil {
+		return clues.Wrap(err, "getting attachments")
+	}
+
+	// Delete unnecessary attachments
+	for _, att := range attachments {
+		name := ptr.Val(att.GetName())
+
+		content, err := api.GetAttachmentContent(att)
+		if err != nil {
+			return clues.Wrap(err, "getting attachment").With("attachment_id", ptr.Val(att.GetId()))
+		}
+
+		found := false
+
+		for _, natt := range event.GetAttachments() {
+			bname := ptr.Val(natt.GetName())
+
+			bcontent, err := api.GetAttachmentContent(natt)
+			if err != nil {
+				return clues.Wrap(err, "getting attachment").With("attachment_id", ptr.Val(natt.GetId()))
+			}
+
+			if name == bname && bytes.Equal(content, bcontent) {
```

Comment:
Is it possible for there to be a delta across names for the same content?  Ie: the attachment name gets updated, but not its content?

Reply:
It is possible that they upload the same file with a different name.

Reply:
But that would be a different attachment.  Is it possible to change an existing attachment name?

Reply:
Nope, changing the name once you upload is not possible.